src/SetupApi/SetupApi.cs

```diff
@@ -219,6 +219,272 @@ public static partial class SetupApi
             uint* requiredSize,
             SetupDiGetDevicePropertyFlags flags);
 
+        /// <summary>
+        /// The <see cref="SetupDiCreateDeviceInfoList(Guid*, IntPtr)"/> function creates an empty device information set and optionally associates the
+        /// set with a device setup class and a top-level window.
+        /// </summary>
+        /// <param name="classGuid">
+        /// A pointer to the <see cref="Guid"/> of the device setup class to associate with the newly created device information set.
+        /// If this parameter is specified, only devices of this class can be included in this device information set. If this parameter is set
+        /// to <see cref="IntPtr.Zero"/>, the device information set is not associated with a specific device setup class.
+        /// </param>
+        /// <param name="hwndParent">
+        /// A handle to the top-level window to use for any user interface that is related to non-device-specific actions (such as a select-device dialog
+        /// box that uses the global class driver list). This handle is optional and can be <see cref="IntPtr.Zero"/>. If a specific top-level window is not
+        /// required, set <paramref name="hwndParent"/> to <see cref="IntPtr.Zero"/>.
+        /// </param>
+        /// <returns>
+        /// The function returns a <see cref="IntPtr"/> to an empty device information set if it is successful. Otherwise, it returns
+        /// <see cref="SafeDeviceInfoSetHandle.Invalid"/>. To get extended error information, call <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/hardware/ff550956(v=vs.85).aspx"/>
+        [DllImport(nameof(SetupApi), SetLastError = true)]
+        public static unsafe extern SafeDeviceInfoSetHandle SetupDiCreateDeviceInfoList(
+            Guid* classGuid,
```

Comment:
Add `[Friendly(FriendlyFlags.In | FriendlyFlags.Optional)]` on this parameter so that we get friendly overloads including (I suspect) `in Guid` so that you can naturally pass a GUID in, while still retaining the ability to pass in `null` on *this* overload.

Reply:
Thanks, I've added these flags. You now get overloads for `Guid*` and `Guid?`, which means that you need to cast `null` to either of these types.

Reply:
> which means that you need to cast null to either of these types.

Ya, that's unfortunate, but not exactly a new problem either in general.

Reply:
nit: style-wise we define a parameter with its attribute on one line.